### PR TITLE
docs(cardinal): Include message refactor

### PR DESCRIPTION
**Pull request by jerargus**
Closes: WORLD-873

## Overview

Update docs to include message registration and iterating over messages.

## Testing and Verifying

No tests

### CODE REVIEW

**Review by smsunarto — CHANGES_REQUESTED** [review]

File: docs/cardinal/game/message.mdx [hunk 1]
```diff
@@ -13,15 +13,15 @@ In Cardinal, transactions are defined using a pair of Go structs representing a
 
 ---
 
-## Defining Messages
+## Using Messages
```
[the remark this answers is not in the document]
Comment (jerargus) [reply docs/cardinal/game/message.mdx]: Done. There was an existing similar example in the systems section. I think having both this targeted example and the more comprehensive example in systems is useful.

File: docs/cardinal/game/message.mdx [hunk 2]
```diff
@@ -3,7 +3,7 @@ title: 'Message'
 description: 'How to define and register a message'
 ---
 
-Messages are predefined user actions that can be submitted to Cardinal and handled within a system. A message is the primary way of representing possible user actions in the game. After a message is executed within a tick, it returns a `Reply` payload that can be read by the game client.
+Messages are predefined user actions that can be submitted to Cardinal and handled within a system. A message is the primary way of representing possible user actions in the game. After a message is processed within a tick, it returns a `Reply` payload that can be read by the game client.
 
 In Cardinal, transactions are defined using a pair of Go structs representing a `Message` and a `Reply`.
```
Comment (technicallyty) [inline docs/cardinal/game/message.mdx:8]: i realize this is not part of the diff, but maybe we can fix this here?
```suggestion
In Cardinal, messages are defined using a pair of Go structs representing a `Message` and a `Reply`.
```

File: docs/cardinal/game/message.mdx [hunk 3]
```diff
@@ -21,7 +21,7 @@ You can easily create a new message and register it to the world by following th
 
 <Steps>
     <Step title="Define the message and reply struct">
-        A message and its reply is defined as a Go struct. You also need to create a singleton message object using the `NewMessageType` function. 
+        A message and its reply is defined as a Go struct.
```
Comment (technicallyty) [inline docs/cardinal/game/message.mdx:24]: thoughts on this addition?
```suggestion
        A message request and its reply is defined as a Go struct.
```

File: docs/cardinal/game/message.mdx (hunk 2 above)
Comment (jerargus) [reply docs/cardinal/game/message.mdx]: done

File: docs/cardinal/game/message.mdx (hunk 3 above)
Comment (jerargus) [reply docs/cardinal/game/message.mdx]: done

**Review by smsunarto — APPROVED** [review]

**Pull request merged**